package.json

```diff
@@ -8,13 +8,16 @@
     "test:watch": "jest --watchAll --colors --passWithNoTests",
     "test": "jest --colors --passWithNoTests",
     "test:ci": "jest --no-colors --passWithNoTests",
-    "prepare": "husky install"
+    "postinstall": "husky install",
+    "pre-publish": "npm run build && shx cp package.json dist/ && shx cp README.md dist/ && cd dist && npm pack",
+    "preversion": "package-check --cwd dist"
   },
-  "main": "./dist/errors/index.js",
-  "types": "./dist/errors/index.d.ts",
+  "main": "./index.js",
+  "types": "./index.d.ts",
   "type": "commonjs",
```

Comment:
`module` field is used by the bundlers and ignored by node js. But bundlers expect entry point in `module` to be an es module(not required, but main purpose of the field is to point to an es module). 
Your entry point is an commonjs module. TS will compile `wrapper.ts` to commonjs module. So, to fix that i suggest to create a wrapper outside of the `errors` directory and use it as entry point for `module` and entry point for `import` condition in `exports` field.

`mjs-wrappers/index.mjs`
```
export * from "./index.js";
```
And copy that wrapper to the dist before packaging.

`.mjs` - it\`s important to have this extension because your `type` field of `package.json` is set to `commonjs` so node js will treat all `.js` files as commonjs modules. But when you add `.mjs` then nodejs will treat them as es modules.

Reply:
Thank you for your review, it's really valuable!